### PR TITLE
bug fixed & added another helper function

### PR DESCRIPTION
**Pull request by g0rchy**
* Fixed a bug where giving a tag to search for, `orunmila` reports back with all records.
* Added a mechanism & another helper function to tells the user whether the tag exists in the db or not.
* Changed a misleading function name & comment (`TagsToString()`) with their appropriate context.

### CODE REVIEW

File: helpers.go [hunk 1]
```diff
@@ -24,12 +24,14 @@ func check(e error) {
 	}
 }
 
-// Convert tag ids into a comma separated string to be used with our query
-func TagsToString() string {
+// Convert tags' ids from the db into a comma separated string
+func TagsToIdsInString() string {
 	var a []string
 	for _, id := range Tags {
+		log.Printf("[TagsToIdsInString] id => %d\n", id)
```
Comment (proditis) [inline helpers.go:31]:
```suggestion
		log.Debugf("[TagsToIdsInString] id => %d\n", id)
```
```suggestion
		log.Printf("[TagsToIdsInString] id => %d\n", id)
```

Comment (proditis) [inline helpers.go:31]:
```suggestion
		log.Debugf("[TagsToIdsInString] id => %d\n", id)
```

File: helpers.go [hunk 2]
```diff
@@ -217,11 +228,19 @@ func searchWordsByTagIds(db *sql.DB, tags string) {
 	populateTagIds(db)
 	removeEmptyTags()
 
-	if len(Tags) > 0 {
-		log.Infoln("Using tags:", Tags)
-		queryStr = fmt.Sprintf(queryStr+" left join wt as t2 on t2.word_id=t1.id WHERE t2.tag_id IN (%s) group by t1.id", TagsToString())
+	if len(tags) > 0 {
```
Comment (proditis) [inline helpers.go:231]: Why not just check if Tags is empty? something like `if len(Tags)>0` instead?

File: helpers.go [hunk 3]
```diff
@@ -209,6 +210,16 @@ func importFileWords(db *sql.DB, filename string) {
 	check(err)
 }
 
+// Search for the needle in the haystack
+func containsValue(haystack map[string]int64, needle string) bool {
```
Comment (proditis) [inline helpers.go:214]: NO `TestContainsValue` in helpers_test.go ?!?!?!? :sob:

**Review by proditis — COMMENTED** [review]
Nice one

File: helpers.go (hunk 2 above)
Comment (g0rchy) [reply helpers.go:231]: when searching in a blank db (with no tags inserted), Tags will have a length of 0, thus printing `No tags were given`, `tags` in the other hand has the user defined tags that we are looking for instead

File: helpers.go [hunk 4]
```diff
@@ -217,11 +228,23 @@ func searchWordsByTagIds(db *sql.DB, tags string) {
 	populateTagIds(db)
 	removeEmptyTags()
 
-	if len(Tags) > 0 {
-		log.Infoln("Using tags:", Tags)
-		queryStr = fmt.Sprintf(queryStr+" left join wt as t2 on t2.word_id=t1.id WHERE t2.tag_id IN (%s) group by t1.id", TagsToString())
+	if len(Tags) == 0 {
+		log.Fatalln("db is empty, insert some words with tags")
+	}
```
Comment (proditis) [inline helpers.go:233]: this breaks orunmila 
```suggestion
	if len(Tags) == 0 {
		log.Fatalln("db is empty, insert some words with tags")
	}
```

**Pull request merged**